pkg/k2/language_host/language_host.go

```diff
@@ -19,15 +21,26 @@ import (
 //go:embed python/python_language_host.py
 var pythonLanguageHost string
 
-type ServerAddress struct {
+type ServerState struct {
 	Log     *zap.SugaredLogger
 	Address string
+	Error   error
+	HasErr  chan error
 	HasAddr chan struct{}
```

Comment:
Might be worth combining both of these channels into
```
Done chan struct{}
```
The `error` in the new HasErr doesn't seem to be used anyway.